src/main/rom.c

```diff
@@ -81,42 +82,51 @@ static int is_valid_rom(const unsigned char *buffer)
         return 0;
 }
 
-/* If rom is a .v64 or .n64 image, byteswap or wordswap loadlength amount of
- * rom data to native .z64 before forwarding. Makes sure that data extraction
- * and MD5ing routines always deal with a .z64 image.
+/* Copies the source block of memory to the destination block of memory while
+ * switching the endianness of .v64 and .n64 images to the .z64 format, which
+ * is native to the Nintendo 64. The data extraction routines and MD5 hashing
+ * function may only act on the .z64 big-endian format.
+ *
+ * IN: src: The source block of memory. This must be a valid Nintendo 64 ROM
+ *          image of 'len' bytes.
+ *     len: The length of the source and destination, in bytes.
+ * OUT: dst: The destination block of memory. This must be a valid buffer for
+ *           at least 'len' bytes.
+ *      imagetype: A pointer to a byte that gets updated with the value of
+ *                 V64IMAGE, N64IMAGE or Z64IMAGE according to the format of
+ *                 the source block. The value is undefined if 'src' does not
+ *                 represent a valid Nintendo 64 ROM image.
  */
-static void swap_rom(unsigned char* localrom, unsigned char* imagetype, int loadlength)
+static void swap_copy_rom(void* dst, const void* src, size_t len, unsigned char* imagetype)
 {
-    unsigned char temp;
-    int i;
-
-    /* Btyeswap if .v64 image. */
-    if(localrom[0]==0x37)
-        {
+    if (*(const uint8_t*) src == 0x37)
```

Comment:
Same here, let the comment to explain what `0x37` is supposed to check: `/* Btyeswap if .v64 image. */`